language/move-binary-format/src/unit_tests/compatibility_tests.rs

```diff
@@ -464,24 +474,295 @@ fn deprecated_add_script_visibility() {
         .is_err());
 }
 
+#[test]
+fn private_entry_to_public_entry_allowed() {
+    let private_module = max_version(mk_module_entry(Visibility::Private as u8, true));
```

Comment:
nit: It seems like you could do the cast to `u8` within the body of `mk_module_entry`?